### PR TITLE
X3d 3d ssfile

### PR DESCRIPTION
**Pull request by kgbudge**
### Background

* We have a use case for transport run on a pure hexahedral X3D mesh.
* While adding the necessary capability to our code library, we found that the X3D_Draco_Mesh_Reader sorts the nodes on a boundary face by node index, destroying the original counterclockwise ordering of the nodes.

### Purpose of Pull Request

* Modify X3D_Draco_Mesh_Reader to store boundary face nodes in the original counterclockwise node ordering, rather than sorted by index.

### Description of changes

* The reader sorts the nodes by node index to allow easy matching of boundary faces to cell faces. This is fine, but the code then added the sorted version to the side to node linkage. The code has been modified to add the original version (listed in counterclockwise order) to the side to node linkage instead.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [ ] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/cdash3/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

**Comment by RyanWollaeger** [comment]
@kgbudge It looks like the diff is picking up changes from previous PRs.  @KineticTheory do you know what's going one here?  Does Github get confused from certain rebase/merges (I see a lot of merge commits in the commit log for this PR)?

**Comment by KineticTheory** [comment]
> 
> 
> @kgbudge It looks like the diff is picking up changes from previous PRs. @KineticTheory do you know what's going one here? Does Github get confused from certain rebase/merges (I see a lot of merge commits in the commit log for this PR)?

Long lived git branches often cause this issue -- especially when there are so many merge commits from `upstream/develop`.  @kgbudge  Can you apply this change set to a new branch based on the current `upstream/develop` and update the PR?

**Comment by KineticTheory** [comment]
> 
> 
> > @kgbudge It looks like the diff is picking up changes from previous PRs. @KineticTheory do you know what's going one here? Does Github get confused from certain rebase/merges (I see a lot of merge commits in the commit log for this PR)?
> 
> Long lived git branches often cause this issue -- especially when there are so many merge commits from `upstream/develop`. @kgbudge Can you apply this change set to a new branch based on the current `upstream/develop` and update the PR?

The diff is clean now.  The changes look good to me. 

@RyanWollaeger Can you review these changes?

**Review by KineticTheory — APPROVED** [review]

File: src/mesh/X3D_Draco_Mesh_Reader.cc [hunk 1]
```diff
@@ -406,7 +406,8 @@ void X3D_Draco_Mesh_Reader::read_bdy_files() {
       if (nodes_in_common == fnode_vec) {
 
         // add to the side-node map
-        x3d_sidenode_map.insert(std::pair<int, std::vector<unsigned>>(num_side, fnode_vec));
+        x3d_sidenode_map.insert(
```
Comment (RyanWollaeger) [inline src/mesh/X3D_Draco_Mesh_Reader.cc:409]: Very nice patch!

**Review by RyanWollaeger — APPROVED** [review]
LGTM

**Comment by KineticTheory** [comment]
I'll merge this if the CI testing passes.

**Pull request merged**